Fix/Feat: CMake builds, Smooth screen movement

KorigamiK opened the pull request:
CMake is more portable and easier to work with so I had to make some changes to get it working.

Adds `screen_down_smooth` and `screen_up_smooth` commands

KorigamiK commented:
@ahrm Please review this PR as I think this will make the packaging and distribution of linux builds much easier.

ahrm commented:
1. Why did you change some header names? E.g. `#include <qkeyevent.h>` to `#include <QKeyEvent>`.
2. Some of the removed headers are actually required (e.g. `#include "pdf_renderer.h"` in `main.cpp`). It might not cause a compile issue because it is included in another included header, but it makes future refactoring of headers more difficult.
3. Removing `#include "fts_fuzzy_match.h"` in `main.cpp` completely breaks windows builds. Generally I recommend not changing headers at all, or if you insist you must make sure the build works on all platforms (linux, macos, windows, android).
4. About `if (font_id == -1) {`: Does this actually happen? It should be impossible for the font not to be available because we include it in the resource files.

KorigamiK commented:
> 1. Why did you change some header names? E.g. `#include <qkeyevent.h>` to `#include <QKeyEvent>`.
> 
>     2. Some of the removed headers are actually required (e.g. `#include "pdf_renderer.h"` in `main.cpp`). It might not cause a compile issue because it is included in another included header, but it makes future refactoring of headers more difficult.
> 
>     3. Removing `#include "fts_fuzzy_match.h"` in `main.cpp` completely breaks windows builds. Generally I recommend not changing headers at all, or if you insist you must make sure the build works on all platforms (linux, macos, windows, android).
> 
>     4. About `if (font_id == -1) {`: Does this actually happen? It should be impossible for the font not to be available because we include it in the resource files.

Thanks for the quick review, I'll make the following changes accordingly

- I changed qt headers because I think newer qt versions (mine is `6.8.1`) I was getting this error for several of them, the new ones are generally what I can find to be correct according to the documentation I find online.
```sh
sioyek/pdf_viewer/input.cpp:8:10: fatal error: 'qkeyevent.h' file not found
    8 | #include <qkeyevent.h>
      |          ^~~~~~~~~~~~~
``` 
1. I understand that you want to explicitly include these headers but these should be included differently if possible. Because they are not used directly I followed clang diagnostic fixes to remove them.
2. I'm not advocating remove these headers but I think the headers that are not required should not be included (especially the c++ std headers.) so I've tried to remove headers that I think won't affect the code. Unfortunately I'm only able to test this on my local linux machine only. Although, this can be move to CI in the future easily
3. Yes, I didn't know that I removed that I think this might have been a mistake. I've fixed this.
4. Which is true but during my testing I was trying to compile sioyek without resources and I experienced a crash due to it. Now, if someone tried to build without the resources he wouldn't not get a crash but an error message instead, which is a little better.

KorigamiK commented on CMakeLists.txt:
```diff
@@ -29,8 +42,6 @@ qt_add_executable(sioyek WIN32 MACOSX_BUNDLE
     pdf_viewer/pdf_renderer.cpp pdf_viewer/pdf_renderer.h
     pdf_viewer/pdf_view_opengl_widget.cpp pdf_viewer/pdf_view_opengl_widget.h
     pdf_viewer/rapidfuzz_amalgamated.hpp
-    pdf_viewer/sqlite3.c pdf_viewer/sqlite3.h
```
This now means that we can now remove `sqlite3.c`, `sqlite3.h` and `sqlite3ext.h` from our codebase. But I'll leave it up to you if we decide to move completely to cmake builds for everything
